src/libertem/udf/base.py

```diff
@@ -400,12 +425,37 @@ def init_task_data(self):
     def init_result_buffers(self):
         self.results = UDFData(self.get_result_buffers())
 
+    def export_results(self):
+        # .. versionadded:: 0.6.0.dev0
+        self.results.export()
+
     def set_meta(self, meta):
         self.meta = meta
 
     def set_slice(self, slice_):
         self.meta.slice = slice_
 
+    def set_backend(self, backend):
+        assert backend in self.get_backends()
+        self._backend = backend
```

Comment:
Hm I'm trying to wrap my head around that one since these methods will also be called on the leader node, right? That means one shouldn't rely on `self.xp` being the same outside of partition scope.

In that case, one could let `self.xp` return np until instructed otherwise. Is that what you had in mind?

Reply:
> will also be called on the leader node, right?

Yes, for getting the dataset-shaped result buffer(s)

>That means one shouldn't rely on self.xp being the same outside of partition scope.

Correct. I think this is mostly important for consistency - if `self.xp` isn't available in `get_result_buffers`, one has to explicitly use numpy, which means when porting a UDF to support GPUs, you have to remember which methods have `xp` available, and which don't. 

> In that case, one could let self.xp return np until instructed otherwise. Is that what you had in mind?

That could work, yes. Don't remember what I had in mind, as it's been a while :grin: